### ISSUE TITLE
-> SPONGEFORGE USERS GO HERE <- Server crash when startup

### ISSUE THREAD

**Issue by avescarelo** [asiekierka/FoamFix#166]
Mod version: 0.10.1-1.12.2
Minecraft version: 1.12.2

The server has crashing, after installing FoamFix

**Error**: Exception generating new chunk

Crash log: https://gist.github.com/pingvikin/0fc777a9448202a16ceb3431aa32602f

**Comment by AsuraTakun** [comment]
Yeah, this is totally a foamfix bug. I just had this issue too. Took me like an hour to realize it is the latest curse release of Foamfix, if you go back to the one released earlier this month it will work fine. So go back to 0.10.0 and it will be fine.

**Comment by asiekierka** [comment]
Sponge compatibility is not promised. Disable general.patchChunkContainer, as it conflicts with a FoamFix fix.

@Aaron1011, how do I best detect Sponge in this situation?

**Comment by ghost** [comment]
@asiekierka I can't find where you mean Disable general.patchChunkContainer as I look both in sponge cfg and Foamfixs config and did not see anything called that

**Comment by gabizou** [comment]
@asiekierka I don't know of a way without potentially causing a class load on a sponge core class (like `SpongeCoreMod` you could query for, but it might be a little expensive at coremod build time since there's no guarantee about coremod initialization. Perhaps @Mumfrey could provide an idea? In short, sponge applies a redirect on `BlockStateContainer` and because asie already modifies the target of the redirect, the mixin fails. I'd doubt it would be possible for you to make that path use Mixins, right?

**Comment by asiekierka** [comment]
@gabizou I am most likely going to be moving to MIxins in 1.13; however, at this time, adding in the Mixin library would inflate FoamFix's size about five to six times, which is not desirable.

**Comment by notfood** [comment]
There is no general.patchChunkContainer option, right now the only way to avoid the crash is to revert to 0.10.0.

**Comment by asiekierka** [comment]
general.patchChunkSerialization. Remembered the name wrong.

**Comment by Aaron1011** [comment]
@asiekierka: You should be able to detect if Sponge is installed by using `LaunchClassloader.getClassBytes` - https://github.com/DimensionalDevelopment/VanillaFix/blob/b6992dc220c65b92f0ad2313ea776271a0b04d45/src/main/java/org/dimdev/vanillafix/ModCompatibilityMixinPlugin.java#L18

**Comment by asiekierka** [comment]
Oke. Good to know!
